Modules/audioop.c

```diff
@@ -170,7 +172,7 @@ st_14linear2ulaw(int16_t pcm_val)       /* 2's complement (14-bit range) */
     if (seg >= 8)           /* out of range, return maximum value. */
         return (unsigned char) (0x7F ^ mask);
     else {
-        uval = (unsigned char) (seg << 4) | ((pcm_val >> (seg + 1)) & 0xF);
+        uval = (unsigned char) (seg * (1 << 4)) | ((pcm_val >> (seg + 1)) & 0xF);
```

Comment:
@matthiasgoergens Now that I have time to look at this properly: I'm not seeing the potential for undefined behaviour here. Isn't `seg` always nonnegative at this point?

Reply:
I'm currently on mobile.

If memory serves right, I just fixed all left-shifts in this file, because there's no harm in the fix when it's not needed, and I didn't trust my fallible reasoning about exactly which shifts and code paths were safe.

I can restrict the change to only those code paths we can verify as unsafe?

Reply:
Basically I went all out on the false positives to avoid any false negative.

Reply:
Thanks, understood. I'd suggest reverting these two changes, though. To "there's no harm in the fix", there's _always_ the risk of unintended consequences, and I think it's worth restricting to the set of fixes that's necessary.